### PR TITLE
Fixes #33: Adds tweet search feature

### LINKED ISSUE #33: Add tweet search feature
When a keyword is searched (using the *magnifying lens* icon in keyboard) in the EditText present in `SuggestFragment` or one of the suggested topic in RecyclerView is clicked, it should navigate to `SearchFragment`. 
- [x] Obtain the JSON data of the search using [search](https://www.loklak.org/api.html#search) and display it in RecyclerView.
- [x] Create a separate RecyclerView adapter to display multiple images in a tweet.
- [x] Show different categories of tweets using `filter` parameter of *search* API endpoint. Tweets to be shown using `TabLayout` and `ViewPager`.
- [ ] Add a floating action button that will redirect to TweetPosting activity (to be added later)

device-2017-06-07-115715

### PR DESCRIPTION
**Pull request by SKrPl**
Related issue: https://github.com/fossasia/loklak_wok_android/issues/33

The search query is obtained from the intent received from
SuggestFragment. Using the "search" API endpoint of loklak tweets are
obtained and inflated. Tweets are displayed in three different
categories i.e. "LATEST", "PHOTOS" and "VIDEOS". This is done by using
"filter" parameter in "search" endpoint and by using TabLayouts and
ViewPager to provide the UI. Boundary cases like:
1. No tweets found for the query and
2. Network error
are handled by changing the visibility of RecyclerView to GONE and the
message to be shown to VISIBLE.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -36,11 +36,12 @@ dependencies {
     compile fileTree(include: ['*.jar'], dir: 'libs')
     compile files('libs/loklakj_lib.jar')
     compile files('libs/android-core.jar')
-
     testCompile 'junit:junit:4.12'
+
     compile 'com.android.support:appcompat-v7:25.3.1'
     compile 'com.android.support:support-v4:25.3.1'
     compile 'com.android.support:design:25.3.1'
+    compile 'com.android.support:cardview-v7:25.3.1'
```
Comment (abishekvashok) [inline app/build.gradle]: Use `implementation` instead of `compile`

**Review by abishekvashok — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (SKrPl) [reply app/build.gradle]: Implementation is not a stable version. So, I think we should stick to compile.

**Comment by SKrPl** [comment]
device-2017-07-11-081432

I haven't added website and video preview as of now, I will create a new issue for that. This PR only solves adds the functionality of tweet searching.

**Comment by SKrPl** [comment]
@mariobehling @Orbiter @sudheesh001 @daminisatya @abishekvashok @Achint08 @djmgit @hemantjadon @kavithaenair @singhpratyush @vibhcool Please review.

**Comment by singhpratyush** [comment]
@SKrPl: The code looks clean and good :+1:. Can you please provide a link to the corresponding APK?

**Comment by SKrPl** [comment]
.apk file link: https://drive.google.com/open?id=0B_L50ue18BeuaUlpTjllalpaZGc

**Comment by Achint08** [comment]
@SKrPl All Images/Videos are not showing up. It is just a tabs implementation, right?

**Comment by SKrPl** [comment]
@Achint08 as of now only the uploaded images and videos will be shown. Using the `filter` parameter of `search` API endpoint provides image tweets that contains website preview and same goes for videos also (youtube). So, this PR doesn't deal with showing website image preview. I will create a separate issue for that and work on that.

File: app/src/main/res/drawable/reply.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<!-- drawable/reply.xml -->
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:height="24dp"
+    android:width="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path android:fillColor="#666666" android:pathData="M10,9V5L3,12L10,19V14.9C15,14.9 18.5,16.5 21,20C20,15 17,10 10,9Z" />
+</vector>
```
Comment (Achint08) [inline app/src/main/res/drawable/reply.xml]: A new line at the end of the file.

File: app/src/main/res/drawable/heart.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<!-- drawable/heart.xml -->
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:height="24dp"
+    android:width="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path android:fillColor="#666666" android:pathData="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z" />
+</vector>
```
Comment (Achint08) [inline app/src/main/res/drawable/heart.xml]: Here also

**Comment by Achint08** [comment]
@SKrPl Switching to a new filter shows up `Please Check your internet connection.` Kindly check this.

**Comment by SKrPl** [comment]
@Achint08 that happens when a NetworkIO exception is encountered. Tapping on the TextView resends the GET request. And the views get populated if there is no network error again.

**Comment by singhpratyush** [comment]
@SKrPl: I see the same picture across many Tweets in the `Image` view. Is there a placeholder?

**Review by Achint08 — APPROVED** [review]
LGTM 👍

**Comment by SKrPl** [comment]
@singhpratyush no there is no placeholder, can you provide a screenshot?

**Comment by singhpratyush** [comment]
@SKrPl: It seems like those Tweets actually had the same image. It is working fine for other queries. Sorry for the false alarm.

**Review by singhpratyush — APPROVED** [review]

**Pull request merged**